File: ProjectSource/RedoCode_backend/src/main/java/com/redocode/backend/Messages/StatisticMessage.java
package com.redocode.backend.Messages;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import lombok.Value;
import lombok.experimental.SuperBuilder;
import lombok.extern.jackson.Jacksonized;

import java.util.List;

@Value
@SuperBuilder
@Jacksonized
@JsonIgnoreProperties(ignoreUnknown = true)
@NoArgsConstructor(force = true)
@AllArgsConstructor
public class StatisticMessage {

  private List<LanguageUsePart> LanguageUse;
  private List<AmountOfLatlyDonePart> amountOfLatelyDone;
}

File: ProjectSource/RedoCode_frontend/src/types/ApiMessages/StatisticMessage.ts
export interface langaugeUsePart {
  name: string
  amount: number
}

export type LanguageUse = langaugeUsePart[]

export interface amountOfLatlyDonePart {
  date: Date
  amount: number
}

export type amountOfLatlyDone = amountOfLatlyDonePart[]

export default interface StatisticMessage {
  languageUse: LanguageUse
  amountOfLatelyDone: amountOfLatlyDone
}
